File: Programming Basics with Python/По-сложни проверки/trade_commissions.py

city = input()
sales = float(input())

if sales < 0:
    print()

total_price = 0
if city == "Sofia":
    if 0 <= sales <= 500:
        total_price = sales * 0.05
    elif 500 < sales <= 1000:
        total_price = sales * 0.07
    elif 1000 < sales <= 10000:
        total_price = sales * 0.08
    elif sales > 10000:
        total_price = sales * 0.12
    else:
        print("error")
elif city == "Varna":
    if 0 <= sales <= 500:
        total_price = sales * 0.045
    elif 500 < sales <= 1000:
        total_price = sales * 0.075
    elif 1000 < sales <= 10000:
        total_price = sales * 0.10
    elif sales > 10000:
        total_price = sales * 0.13
    else:
        print("error")
elif city == "Plovdiv":
    if 0 <= sales <= 500:
        total_price = sales * 0.055
    elif 500 < sales <= 1000:
        total_price = sales * 0.08
    elif 1000 < sales <= 10000:
        total_price = sales * 0.12
    elif sales > 10000:
        total_price = sales * 0.145
    else:
        print("error")
else:
    print("error")

if total_price != 0:
    print(f"{total_price:.2f}")

File: Programming Basics with Java/More complex checks/TradeCommissions.java
import java.util.Scanner;

public class TradeCommissions 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        String city = input.nextLine();
        double sales = input.nextDouble();

        boolean parameter = true;
        double commission = 0.0;

        switch (city)
        {
            case "Sofia":
            {
                if (0 <= sales && sales <= 500)
                {
                    commission = 0.05;
                }
                else if (sales <= 1000)
                {
                    commission = 0.07;
                }
                else if (sales <= 10_000)
                {
                    commission = 0.08;
                }
                else if (sales > 10_000)
                {
                    commission = 0.12;
                }
                else
                {
                    parameter = false;
                }
                break;
            }
            case "Varna":
            {
                if (0 <= sales && sales <= 500)
                {
                    commission = 0.045;
                }
                else if (sales <= 1000)
                {
                    commission = 0.075;
                }
                else if (sales <= 10_000)
                {
                    commission = 0.1;
                }
                else if (sales > 10_000)
                {
                    commission = 0.13;
                }
                else
                {
                    parameter = false;
                }
                break;
            }
            case "Plovdiv":
            {
                if (0 <= sales && sales <= 500)
                {
                    commission = 0.055;
                }
                else if (sales <= 1000)
                {
                    commission = 0.08;
                }
                else if (sales <= 10_000)
                {
                    commission = 0.12;
                }
                else if (sales > 10_000)
                {
                    commission = 0.145;
                }
                else
                {
                    parameter = false;
                }
                break;
            }
            default:
            {
                parameter = false;
            }
        }

        if (parameter && sales >= 0)
        {
            System.out.printf("%.2f", sales * commission);
        }
        else
        {
            System.out.println("error");
        }
    }
}
